src/backend/distributed/worker/worker_partition_protocol.c

```diff
@@ -110,12 +111,23 @@ worker_range_partition_table(PG_FUNCTION_ARGS)
 	uint32 taskId = PG_GETARG_UINT32(1);
 	text *filterQueryText = PG_GETARG_TEXT_P(2);
 	text *partitionColumnText = PG_GETARG_TEXT_P(3);
+	char *partitionColumnNameCandidate = text_to_cstring(partitionColumnText);
+
+	char *partitionColumnName = NULL;
+	char *endptr = NULL;
+	uint32 partitionColumnIndex = strtoul(partitionColumnNameCandidate, &endptr,
+										  10 /*base*/);
+	if (endptr == partitionColumnNameCandidate)
```

Comment:
this could use a nice and big comment on why we'd do something so crazy :)

Reply:
@marcocitus added a comment